README.md

```diff
@@ -13,3 +13,25 @@
+2. Within asymptotic complexity, we find the fastest growing element, and usually ignore low growing and constant factors. While this may work for many data sets, as the data increases, the runtime can be more than the asymptotic complexity rounded to. Since this is theoretically true, in practice, we cannot always ignore constant factors since they still do contribute to the runtime, even if at a small volume.
+3. Depending on the machines the code runs on, the same code can vary in time. Asymptotic analysis does not take in external variables.
+
+Part 2:
+Within binary trees, they usually have a complexity of log2(n). So, to find the estimated runtime we can find this by inputting our values. For 1000 elements the runtime is estimated at log2(1000) = 9.9 seconds. For 10000 elements, the runtime is estimated at log2(10000) = 13.3 seconds. This is just an estimation. To see how the logarithmic graph grows to these two points, we can divide them to see how much they grow, (13.3/9.9) = 1.34. So, now taking into account the runtime we are given for 1000 elements which is 5 seconds, we can multiply our rate that goes from 1000 to 10000 by the number resulting in 5 * 1.34 = 6.7 seconds. 
+
+Part 3:
+1. One reason is the cache. With larger data sets, cache misses are more frequent. When the cache fills up, data is moved to slower memory, and the program may struggle to find it, leading to slower performance. While this can happen with smaller data sets, it’s more common with larger ones due to the increased likelihood of filling the cache.
+2. Depending on the implementation, if the code is very very ineffcient, the time can become much larger. 
+3. An unbalanced binary tree can cause performance issues as the data set grows. While smaller data sets may see minimal runtime differences, larger ones can significantly impact performance. Asymptotic analysis can be misleading, as it may not account for the effects of inserting large data into an unbalanced tree. Worst case could even be a binary tree that is so unbalaced it resembles a linked list, and if the result is at the bottom, this can definitely cause runtime errors. The worst runtime you could expect is if it had to go through every node, which would be O(n). The runtime would be O(10000), which may account for why in the example it took 100 seconds.
```

Comment:
How long would you expect it to take if you assume linear complexity? Please answer by giving the number of seconds you expect it to take.